### PR TITLE
Remove locations by_type and by_name views

### PR DESCRIPTION
**Pull request by esoergel**
@nickpell @czue
Should be good to go.  The views aren't used anywhere (except for that ilsgateway call, which has been removed).

### CODE REVIEW

File: corehq/apps/locations/tests/util.py [hunk 1]
```diff
@@ -17,16 +16,12 @@ def make_loc(code, name=None, domain=TEST_DOMAIN, type=TEST_LOCATION_TYPE, paren
...
-            iter_bulk_delete(model.get_db(), ids)
-        except BulkSaveError:
-            pass
+    ids = [
+        doc['id'] for doc in
+        SupplyPointCase.get_db().view('commtrack/supply_point_by_loc', reduce=False).all()
+    ]
+    iter_bulk_delete(SupplyPointCase.get_db(), ids)
+
+    iter_bulk_delete(Location.get_db(), SQLLocation.objects.location_ids())
```
Comment (esoergel) [inline corehq/apps/locations/tests/util.py]: Get the list of couch location ids from SQL

File: custom/ilsgateway/tanzania/handlers/register.py [hunk 2]
```diff
@@ -28,18 +28,11 @@ def _get_facility_location(self, domain, msd_code):
...
-                                 startkey=[domain, "DISTRICT", sp],
-                                 endkey=[domain, "DISTRICT", sp],
-                                 reduce=False,
-                                 include_docs=True)
-        return locs[0]
+        return SQLLocation.objects.filter(
+            domain=domain,
+            location_type__name="DISTRICT",
+            name=sp,
+        )[0].couch_location
```
Comment (esoergel) [inline custom/ilsgateway/tanzania/handlers/register.py]: @lwyszomi @kkrampa can you confirm that this is equivalent?  I'm removing couch location views and switching things to SQL.

Comment (kkrampa) [inline custom/ilsgateway/tanzania/handlers/register.py]: @esoergel Eventually this file won't be used so for me your change is ok.

Comment (esoergel) [inline custom/ilsgateway/tanzania/handlers/register.py]: Sounds good, thanks for the update.  If at some point this code stops being used, could you remove the unused code?  Every line of code has maintenance costs associated with it, such as this substitution I've done here, so we should liberally remove unneeded code.

**Comment by esoergel** [comment]
I just offlined with Danny - this will require a reindex because there are other views in the `_design` directory.  In light of that, I'd like to hold off on deploying this until we can also deploy the removal of the `prop_index_site_code` view.  That's currently blocking on the merge of https://github.com/dimagi/commcare-hq/pull/7128.

Could someone please review and sign off on this PR, then I can handle merging both of them together whenever appropriate?

**Comment by nickpell** [comment]
signing off

**Comment by czue** [comment]
kicked build. :+1: when it passes and you're ready to merge.

**Comment by dannyroberts** [comment]
@esoergel did you want to reindex this over the weekend? If probably want to merge right after today's deploy

**Comment by esoergel** [comment]
@dannyroberts Not quite yet, I'm also trying to get rid of one more view, and I want to get https://github.com/dimagi/commcare-hq/pull/7160 live first for that.  I don't think it's worth doing two separate reindexes, as this isn't particularly urgent.

**Comment by dannyroberts** [comment]
Ah, gotcha, thanks. Great!

**Comment by esoergel** [comment]
@dannyroberts This has been stable for long enough now that I'm ready to merge.  Think 1 day is long enough?  I can merge after today's deploy then kick off a preindex.

**Comment by dannyroberts** [comment]
Sure sounds good!

**Comment by dannyroberts** [comment]
@esoergel deploy successful so I think you can go ahead and merge and preindex @TylerSheffels FYI

**Pull request merged**